src/ews_xsd.erl

```diff
@@ -383,54 +481,88 @@ to_string(Val) -> Val.
+            process(Rest, Ts, TypeAcc, [Elem | ElemAcc], TypeMap, Model)
+    end;
+process([#element{parts=[{doc, _}]} = E | Rest], Ts, TypeAcc, ElemAcc,
+        TypeMap, Model) ->
+    process([E#element{parts=[]} | Rest], Ts, TypeAcc, ElemAcc, TypeMap, Model);
+process([#element{name=_Name, type=#reference{name=Qname}, parts=[]} = _E | Rest], Ts,
+        TypeAcc, ElemAcc, TypeMap, Model) ->
+    %% io:format("Ref: ~p~n", [E]),
+    case ews_model:get_elem(Qname, TypeMap) of
+        false ->
+            error({cant_find_in_typemap, Qname});
+        #elem{type = #base{}} = E1 ->
+            process(Rest, Ts, TypeAcc, [E1 | ElemAcc], TypeMap, Model)
+    end;
 process([#element{name=Qname, type=T, parts=[]} = E | Rest], Ts,
-        TypeAcc, ElemAcc) ->
+        TypeAcc, ElemAcc, TypeMap, Model) ->
     Meta = parse_meta(E),
     Qtype = qname(T, no_ns),
-    case to_base(T) of
+    case to_base(Qtype) of
         false ->
             case lists:keyfind(Qtype, 1, Ts) of
                 false ->
                     Elem = #elem{qname=Qname, type=Qtype, meta=Meta},
-                    process(Rest, Ts, TypeAcc, [Elem | ElemAcc]);
+                    ews_model:put(Elem, Model, TypeMap),
+                    process(Rest, Ts, TypeAcc, [Elem | ElemAcc], TypeMap, Model);
                 {Qtype, BaseOrEnum} ->
                     Elem = #elem{qname=Qname, type=BaseOrEnum, meta=Meta},
-                    process(Rest, Ts, TypeAcc, [Elem | ElemAcc])
+                    ews_model:put(Elem, Model, TypeMap),
+                    process(Rest, Ts, TypeAcc, [Elem | ElemAcc], TypeMap, Model)
             end;
         #base{} = Base ->
             Elem = #elem{qname=Qname, type=Base, meta=Meta} ,
-            process(Rest, Ts, TypeAcc, [Elem | ElemAcc])
+            ews_model:put(Elem, Model, TypeMap),
+            process(Rest, Ts, TypeAcc, [Elem | ElemAcc], TypeMap, Model)
     end;
-process([#simple_type{} | Rest], Ts, TypeAcc, ElemAcc) ->
-    process(Rest, Ts, TypeAcc, ElemAcc);
+process([#simple_type{} | Rest], Ts, TypeAcc, ElemAcc, TypeMap, Model) ->
+    process(Rest, Ts, TypeAcc, ElemAcc, TypeMap, Model);
 process([#complex_type{name=Qname, extends=Ext, parts=Ps} | Rest], Ts,
-        TypeAcc, ElemAcc) ->
-    {AccWithSubTypes, SubElems} = process(Ps, Ts, TypeAcc, []),
+        TypeAcc, ElemAcc, TypeMap, Model) ->
+    {AccWithSubTypes, SubElems} = process(Ps, Ts, TypeAcc, [], TypeMap, Model),
     Type = #type{qname=Qname, extends=Ext, elems=SubElems},
-    process(Rest, Ts, [Type | AccWithSubTypes], ElemAcc);
-process([_T | Rest], Ts, TypeAcc, ElemAcc) ->
-    %% io:format("error: unexpected ~p~n", [T]),
-    process(Rest, Ts, TypeAcc, ElemAcc);
-process([], _, TypeAcc, ElemAcc) ->
+    ews_model:put(Type, Model, TypeMap),
+    process(Rest, Ts, [Type | AccWithSubTypes], ElemAcc, TypeMap, Model);
+process([T | Rest], Ts, TypeAcc, ElemAcc, TypeMap, Model) ->
+    io:format("error: unexpected ~p~n", [T]),
```

Comment:
Same as above, remove or put behind -ifdef.

Reply:
This should probably be here since it is an actual error.

Reply:
Although, should it be a logger:error then?